src/tron.jl

```diff
@@ -1,27 +1,72 @@
 #  Some parts of this code were adapted from
 # https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/optimize/tron.py
 
-export tron
+export tron, TronSolver
 
 tron(nlp::AbstractNLPModel; variant = :Newton, kwargs...) = tron(Val(variant), nlp; kwargs...)
 
 """
-    tron(nlp)
+    tron(nlp; kwargs...)
 
----
+A pure Julia implementation of a trust-region solver for bound-constrained optimization:
+    
+        min f(x)    s.t.    ℓ ≦ x ≦ u
+    
+For an advanced usage, one can first define an `TronSolver` preallocating the memory used in the algorithm and then call `solve!`.
 
     solver = TronSolver(nlp)
-    output = solve!(solver, nlp)
+    solve!(solver, nlp; kwargs...)
+
+# Arguments
+- `nlp::AbstractNLPModel` represents the model solved, see `NLPModels.jl`.
+The keyword arguments may include
+- `subsolver_logger::AbstractLogger = NullLogger()`: subproblem's logger.
+- `x = nlp.meta.x0`: the initial guess.
+- `μ₀::T = T(1e-2)`: algorithm's parameter.
+- `μ₁::T = one(T)`: algorithm's parameter.
+- `σ::T = T(10)`: algorithm's parameter.
+- `max_eval::Int = -1`: maximum number of objective function evaluations.
+- `max_time::Float64 = 30.0`: maximum time limit.
+- `max_cgiter::Int = 50`: subproblem's iteration limit.
+- `use_only_objgrad::Bool = false`: If `true`, the algorithm uses only the function `objgrad` instead of `obj` and `grad`.
+- `cgtol::T = T(0.1)`: subproblem's tolerance.
```

Comment:
initial tolerance?

Reply:
Thanks you for the suggestions!

I am not sure what you mean by initial here. The parameter `cgtol` doesn't seem to be updated.

Reply:
It was supposed to be updated. Must have changed in some iteration.

Reply:
I checked the file history, and I haven't found a modification of this. Should we modify the docstring with
```
- `cgtol::T = T(0.1)`: subproblem's initial tolerance.
``` 
and mark this as an issue to be investigated?

Reply:
I agree with that solution

Reply:
Maybe this parameter isn't updated.